### PR TITLE
[WIP] Remove remaining deprecated interfaces

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

This PR tracks all the places where I found something that may be removed.
Please anyone, take inspiration from this patch and make it in your own PR.
I'll keep rebasing this one until it's empty.

### CODE REVIEW

File: src/Symfony/Component/DomCrawler/Crawler.php [hunk 1]
```diff
@@ -901,10 +901,7 @@ private function relativize($xpath)
                 $expression = substr($expression, strlen($parenthesis));
             }
 
-            // BC for Symfony 2.4 and lower were elements were adding in a fake _root parent
-            if (0 === strpos($expression, '/_root/')) {
```
Comment (stof) [inline src/Symfony/Component/DomCrawler/Crawler.php]: We should add a deprecation warning for this in 2.8 first

File: src/Symfony/Component/Form/Extension/Core/Type/ChoiceType.php [hunk 2]
```diff
@@ -299,7 +291,6 @@ public function configureOptions(OptionsResolver $resolver)
         $resolver->setDefaults(array(
             'multiple' => false,
             'expanded' => false,
-            'choice_list' => null, // deprecated
             'choices' => array(),
             'choices_as_values' => false,
```
Comment (stof) [inline src/Symfony/Component/Form/Extension/Core/Type/ChoiceType.php]: this option should be deprecated saying it has no effect anymore (and the value should be ``true`` to reflect the behavior of the code).
The option should be kept though, to allow writing code compatible with 2.8 and 3.0 (passing the value explicitly)

File: src/Symfony/Component/Form/Extension/Csrf/Type/FormTypeCsrfExtension.php [hunk 3]
```diff
@@ -121,12 +121,10 @@ public function finishView(FormView $view, FormInterface $form, array $options)
      */
     public function configureOptions(OptionsResolver $resolver)
     {
-        // BC clause for the "intention" option
         $csrfTokenId = function (Options $options) {
             return $options['intention'];
```
Comment (stof) [inline src/Symfony/Component/Form/Extension/Csrf/Type/FormTypeCsrfExtension.php]: intention should be removed entirely actually

File: src/Symfony/Component/Form/Extension/Csrf/Type/FormTypeCsrfExtension.php [hunk 4]
```diff
@@ -121,12 +121,10 @@ public function finishView(FormView $view, FormInterface $form, array $options)
...
         };
 
-        // BC clause for the "csrf_provider" option
         $csrfTokenManager = function (Options $options) {
             if ($options['csrf_provider'] instanceof CsrfTokenManagerInterface) {
```
Comment (stof) [inline src/Symfony/Component/Form/Extension/Csrf/Type/FormTypeCsrfExtension.php]: ``csrf_provider`` should be removed entirely

File: src/Symfony/Component/Form/Extension/Validator/EventListener/ValidationListener.php [hunk 5]
```diff
@@ -57,7 +57,6 @@ public function validateForm(FormEvent $event)
             foreach ($violations as $violation) {
                 // Allow the "invalid" constraint to be put onto
                 // non-synchronized forms
-                // ConstraintViolation::getConstraint() must not expect to provide a constraint as long as Symfony\Component\Validator\ExecutionContext exists (before 3.0)
                 $allowNonSynchronized = (null === $violation->getConstraint() || $violation->getConstraint() instanceof Form) && Form::NOT_SYNCHRONIZED_ERROR === $violation->getCode();
```
Comment (stof) [inline src/Symfony/Component/Form/Extension/Validator/EventListener/ValidationListener.php]: the null check can be removed now., not only the comment

Comment (stof) [inline src/Symfony/Component/Form/Extension/Validator/EventListener/ValidationListener.php]: well, not really though, as the component still supports Validator 2.8. This means the comment should be kept too

File: src/Symfony/Component/Validator/Constraints/CollectionValidator.php [hunk 6]
```diff
@@ -42,10 +42,6 @@ public function validate($value, Constraint $constraint)
         // Since the context of the validator is overwritten when initialize()
         // is called for the nested constraint, the outer validator is
         // acting on the wrong context when the nested validation terminates.
-        //
-        // A better solution - which should be approached in Symfony 3.0 - is to
-        // remove the initialize() method and pass the context as last argument
-        // to validate() instead.
```
Comment (stof) [inline src/Symfony/Component/Validator/Constraints/CollectionValidator.php]: this should be kept

File: src/Symfony/Component/Form/Extension/Core/Type/ChoiceType.php [hunk 7]
```diff
@@ -320,11 +311,9 @@ public function configureOptions(OptionsResolver $resolver)
             'choice_translation_domain' => true,
         ));
 
-        $resolver->setNormalizer('choice_list', $choiceListNormalizer);
```
Comment (Tobion) [inline src/Symfony/Component/Form/Extension/Core/Type/ChoiceType.php]: `$choiceListNormalizer` still exists and is unused

Comment (nicolas-grekas) [inline src/Symfony/Component/Form/Extension/Core/Type/ChoiceType.php]: removed

File: src/Symfony/Component/Form/Extension/Core/Type/ChoiceType.php [hunk 8]
```diff
@@ -299,9 +277,8 @@ public function configureOptions(OptionsResolver $resolver)
         $resolver->setDefaults(array(
             'multiple' => false,
             'expanded' => false,
-            'choice_list' => null, // deprecated
             'choices' => array(),
-            'choices_as_values' => false,
+            'choices_as_values' => true,
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Form/Extension/Core/Type/ChoiceType.php]: Comment from @stof:  this option should be deprecated saying it has no effect anymore (and the value should be true to reflect the behavior of the code). The option should be kept though, to allow writing code compatible with 2.8 and 3.0 (passing the value explicitly).

**Comment by nicolas-grekas** [comment]
@stof would you like to take over this PR? I'm not conformable enough with the code there...

**Pull request closed without merging**